src/drawing/DrawingList.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import classNames from 'classnames';
+import cloneDeep from 'lodash/cloneDeep';
+import noop from 'lodash/noop';
+import DrawingTarget from './DrawingTarget';
+import useOutsideEvent from '../common/useOutsideEvent';
+import { AnnotationDrawing, PathGroup, Shape } from '../@types';
+import { checkValue } from '../utils/util';
+import { getShape } from './drawingUtil';
+
+export type Props = {
+    activeId?: string | null;
+    annotations: AnnotationDrawing[];
+    className?: string;
+    onSelect?: (annotationId: string | null) => void;
+};
+
+export const DRAWING_TARGET_PADDING = 10;
+
+export function filterDrawing({ target: { path_groups: pathGroups } }: AnnotationDrawing): boolean {
+    return pathGroups.reduce<boolean>(
+        (prevGroups, { paths }) =>
+            prevGroups &&
+            paths.reduce<boolean>(
+                (prevPaths, { points }) =>
+                    prevPaths &&
+                    points.reduce<boolean>(
+                        (prevPoints, { x, y }) => prevPoints && checkValue(x) && checkValue(y),
```

Comment:
This is pretty deeply nested (arrow code). We should try to flatten it out, if possible.

Reply:
`reduce` is changed to `every` now, which makes the code much cleaner.